iothub/service/src/Http2/HttpRequestMessageFactory.cs

```diff
@@ -40,12 +40,13 @@ public HttpRequestMessageFactory(Uri baseUri, string apiVersion)
         /// <param name="relativeUri">The URI that the request will be made to.</param>
         /// <param name="authorizationProvider">The provider of authorization tokens.</param>
         /// <param name="payload">The payload for the request to be serialized. If null, no payload will be in the request.</param>
+        /// <param name="extraParam">Any extra parameters to be added to request URI</param>
         /// <returns>The created HTTP request.</returns>
-        internal HttpRequestMessage CreateRequest(HttpMethod method, Uri relativeUri, IotHubConnectionProperties authorizationProvider, object payload = null)
+        internal HttpRequestMessage CreateRequest(HttpMethod method, Uri relativeUri, IotHubConnectionProperties authorizationProvider, object payload = null, Uri extraParam = null)
         {
             var message = new HttpRequestMessage();
             message.Method = method;
-            message.RequestUri = new Uri(_baseUri, relativeUri.ToString() + _apiVersionQueryString);
+            message.RequestUri = new Uri(_baseUri, relativeUri.ToString() + _apiVersionQueryString + (extraParam == null ? "" : extraParam.ToString()));
```

Comment:
If extraParam is either ignored or cast as a string, then why are you accepting it as a URI?